### PR TITLE
Add xml_file_path option to Neuroscope

### PR DESCRIPTION
**Pull request by h-mayorquin**
The neo reader for Neuroscope supports indicating the xml_file_path that is used to extract the metadata. The option, however, has not been propagated to spikeinterface. This PR solves that.

### CODE REVIEW

File: src/spikeinterface/extractors/neoextractors/neuroscope.py [hunk 1]
```diff
@@ -43,16 +45,25 @@ class NeuroScopeRecordingExtractor(NeoBaseRecordingExtractor):
...
-        neo_kwargs = {"filename": str(file_path)}
+    def map_to_neo_kwargs(cls, file_path, xml_file_path=None):
+        # For this because of backwards compatibility we have a strange convention
+        # filename is the xml file
+        # binary_file is the binary file in .dat, .lfp, .eeg
+
+        if xml_file_path is not None:
+            neo_kwargs = {"binary_file": file_path, "filename": xml_file_path}
+        else:
+            neo_kwargs = {"filename": file_path}
```
Comment (CodyCBakerPhD) [inline src/spikeinterface/extractors/neoextractors/neuroscope.py:65]: o.O

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin LGTM, but can you add test(s)?

File: src/spikeinterface/extractors/neoextractors/neuroscope.py (hunk 1 above)
Comment (h-mayorquin) [reply src/spikeinterface/extractors/neoextractors/neuroscope.py:65]: Yeah, the API of neo is showing age here. I tried to document it better on that side:

https://github.com/NeuralEnsemble/python-neo/pull/1290

File: src/spikeinterface/extractors/tests/common_tests.py [hunk 2]
```diff
@@ -38,13 +39,6 @@ def test_open(self):
             # test streams and blocks retrieval
             full_path = self.get_full_path(path)
 
-            extractor_name = self.ExtractorClass.name
-            print(f"Extractor name {extractor_name} - path {full_path}")
-            nblocks = get_neo_num_blocks(extractor_name, full_path)
-            stream_names, stream_ids = get_neo_streams(extractor_name, full_path)
-
-            print(f"Num blocks: {nblocks}, Stream names: {stream_names}, Stream IDs: {stream_ids}")
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/tests/common_tests.py:46]: @alejoe91 I am removing this code that seems to be some kind of debugging.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by h-mayorquin** [comment]
@alejoe91 @CodyCBakerPhD Your approval is somehow not working for this. I am not sure if it did before.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
GitHub says

> At least 1 approving review is required by reviewers with write access. 

which I don't know if I have (I don't think so though) so my approval is merely symbolic of my appreciation of your efforts 😆

**Comment by h-mayorquin** [comment]
Greatly appreciated though!

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**